src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -364,6 +364,20 @@ void maybeAddPartitions(int minPartitionNum) throws ExecutionException, Interrup
       return newPartitionAssignments;
     }
 
+    private static BrokerMetadata randomBroker(Set<BrokerMetadata> brokers) {
+      int brokerSetSize = brokers.size();
+      // In practicality, the Random object should be rather more shared than this.
+      int random = new Random().nextInt(brokerSetSize);
+      int index = 0;
+      for (BrokerMetadata brokerMetadata : brokers) {
+        if (index == random)
+          return brokerMetadata;
+        index++;
+      }
+
+      throw new IllegalStateException("Couldn't find random broker.");
```

Comment:
Consider checking empty and null `brokers` argument at the top so that returning random `BrokerMetadata` is guaranteed. Also, `IllegalArgumentException`.

Reply:
will address your concerns in a sep. PR.

Reply:
https://github.com/linkedin/kafka-monitor/pull/264